packages/camera/camera/lib/src/camera_controller.dart

```diff
@@ -482,6 +485,32 @@ class CameraController extends ValueNotifier<CameraValue> {
     }
   }
 
+  /// TODO: Documentation
+  void onCameraTimeLimitReachedEvent({onCameraTimeLimitReached}) {
+    if (!value.isInitialized || _isDisposed) {
+      throw CameraException(
+        'Uninitialized CameraController',
+        'cameraTimeLimitReachedEventStream was called on uninitialized CameraController',
+      );
+    }
+    if (!value.isRecordingVideo) {
+      throw CameraException(
+        'No video is recording',
+        'cameraTimeLimitReachedEventStream was called when no video is recording.',
+      );
+    }
+    debugPrint('ping');
+
+    CameraPlatform.instance.onCameraTimeLimitReached(_cameraId).listen((event) {
```

Comment:
Doesn't get closed after receiving an event, onCameraTimeLimitReached gets printed multiple times in 2nd+ recording.